src/com/reco1l/osu/hud/GameplayHUD.kt

```diff
@@ -0,0 +1,276 @@
+
+        if (layoutData == HUDSkinData.Default) {
+            applyDefaultLayout()
+        }
+    }
+
+    private inline fun <reified T : HUDElement>getFirstOf() : T? {
+        return mChildren?.firstOrNull { it is T } as? T
+    }
+
+    private fun applyDefaultLayout() {
+        // Default layout is hardcoded to keep the original layout of the game before the HUD
+        // editor was implemented. As well there's no other way since the original layout was
+        // using cross references between elements that are not possible to be set in the editor.
+
+        val scoreCounter = getFirstOf<HUDScoreCounter>()
+        scoreCounter?.anchor = Anchor.TopRight
+        scoreCounter?.origin = Anchor.TopRight
+        scoreCounter?.setScale(0.96f)
+        scoreCounter?.x = -10f
+
+        val accuracyCounter = getFirstOf<HUDAccuracyCounter>()
+        accuracyCounter?.anchor = Anchor.TopRight
+        accuracyCounter?.origin = Anchor.TopRight
+        accuracyCounter?.setScale(0.6f * 0.96f)
+        accuracyCounter?.setPosition(-17f, 9f)
+
+        if (scoreCounter != null && accuracyCounter != null) {
+            accuracyCounter.y += scoreCounter.y + scoreCounter.drawHeight
+        }
+
+        val pieSongProgress = getFirstOf<HUDPieSongProgress>()
+        pieSongProgress?.anchor = Anchor.TopRight
+        pieSongProgress?.origin = Anchor.CenterRight
+
+        if (pieSongProgress != null && accuracyCounter != null) {
+            pieSongProgress.y = accuracyCounter.y + accuracyCounter.heightScaled / 2f
+            pieSongProgress.x = accuracyCounter.x - accuracyCounter.widthScaled - 18f
+        }
+
+        val comboCounter = getFirstOf<HUDComboCounter>()
+        comboCounter?.anchor = Anchor.BottomLeft
+        comboCounter?.origin = Anchor.BottomLeft
+        comboCounter?.setPosition(10f, -10f)
+        comboCounter?.setScale(1.28f)
+    }
+
+    //endregion
+
+    //region Elements events
+    fun setEditMode(value: Boolean) {
+        isInEditMode = value
+        GlobalManager.getInstance().gameScene.isHUDEditorMode = value
+
+        if (value) {
+            ResourceManager.getInstance().loadHighQualityAsset("delete", "delete.png")
+            ResourceManager.getInstance().loadHighQualityAsset("scale", "scale.png")
+            ResourceManager.getInstance().loadHighQualityAsset("rotate", "rotate.png")
+            ResourceManager.getInstance().loadHighQualityAsset("rotate_left", "rotate_left.png")
+            ResourceManager.getInstance().loadHighQualityAsset("rotate_right", "rotate_right.png")
+
+            elementSelector = HUDElementSelector(this)
+
+            parent!!.attachChild(elementSelector)
+            parent!!.registerTouchArea(elementSelector)
+
+        } else {
+            parent!!.detachChild(elementSelector)
+            parent!!.unregisterTouchArea(elementSelector)
+
+            elementSelector = null
+        }
+
+        mChildren?.filterIsInstance<HUDElement>()?.forEach { it.setEditMode(value) }
```

Comment:
Can use `forEachElement` since this doesn't mutate the list.

Reply:
That function doesn't exists...

Reply:
https://github.com/osudroid/osu-droid/blob/35ac6569e88daf97f7d5f8f333dfbedf57b62534/src/com/reco1l/osu/hud/GameplayHUD.kt#L231-L235

```suggestion
        forEachElement { it.setEditMode(value) }
```